typing/jkind.mli

```diff
@@ -179,31 +179,41 @@ end
 
 module Const : sig
   (** Constant jkinds are used for user-written annotations *)
-  type t = Types.type_expr Jkind_types.Const.t
+  type 'd t = (Types.type_expr, 'd) Jkind_types.Const.t
```

Comment:
I'm not sure why we want to index constants over allowance. I thought constants are alwasy LR. Let's discuss offline.

Reply:
We indeed discussed, and I believe @riaqn understands the motivation better. In short: `Jkind.Const.t` is *not* constant. It simply doesn't have any sort variables, but it still might have `with 'a` and such. @riaqn resolve if you're satisfied here.

Reply:
Yes it's constant in terms of sort variables (which is not described by allowance anyway). It's not constant in terms of mode crossing. We agree that further refactoring will clean up this bit, so it's fine.

Could you add a comment here to clarify that?